serving/docs/lmi/tutorials/trtllm_finding_max_num_tokens_tutorial.md

```diff
@@ -0,0 +1,125 @@
+# TensorRT-LLM finding max_num_tokens tutorial
+
+From LMI 0.27.0 (TRTLLM 0.8.0), we introduced `max_num_tokens` build option to simplify the compilation of TRTLLM model as well as supporting all kinds of traffic pattern.
+This max number of tokens represent the shared tokens across the TRTLLM framework regardless of the batch size or context length.
+It is similar to pagedAttention tokens (e.g max_batched_tokens) settings in vLLM. If you set this value in LMI, it is equivalent to the followings:
+
+```
+Serving prperties: option.max_num_tokens=50000
+ENV: OPTION_MAX_NUM_TOKENS=50000
+```
+
+It is the same as:
```

Comment:
I feel like this is a bit confusing, since 4096 * 256 > 50000. They're not really the same

I think we should call out that max_input_len, max_output_len, and max_rolling_batch_size are now used as client controls (limits the types of requests clients can send, and the maximum number in the batch). Without any of these restrictions specified, we basically allow any composition of requests as long as we do not exceed 50000 tokens total across all requests in a single batch

Reply:
If user does not specify `max_num_tokens`:
- do we automatically set `max_num_tokens` to `batch_size * (max_input_len + max_output_len)`? I assume we also have default values for each of those in case user does not specify?

If user only specifies `max_num_tokens`, do we have defaults for batch_size, input/output len as well that would constrain the types of requests clients can send?

Reply:
Good question, if user does not specify max_num_tokens, max_num_tokens is not automatically set - we then use max_batch_size, input len, and output len. Those have defaults - I believe it is batch size 16, input 1024, output 512. That would definitely set limits on the input and output length of the request

Reply:
If user not specify max_num_tokens, we simply not setting them and fallback to the old ways. User will going to suffer from the input/output and batch size limitation. I think this is a good call out, we can try to make a max_num_tokens calculated if customer specify the remaining three. So we can always support batch size 256